### PR TITLE
[installer]: validate the AuthProvider config value

### LINKED ISSUE #6750: [installer]: configure AuthProviders settings
In call with @kylos101, we found out the issue with the `authProviders` array. Missing the `id`, `host` and `type` from the `AuthProviderConfigs`

### PR DESCRIPTION
**Pull request by MrSimonEmms**
## Description

Gets the auth provider config working. Adds the id, host and type params to the AuthProvider and also adds validation on the required fields

## Related Issue(s)

Fixes #6750

## How to test

Run installer with config:

```yaml
authProviders:
  - id: Public-GitHub
    host: github.com
    type: GitHub
    oauth:
      clientId: xxx
      clientSecret: xxx
      callBackUrl: https://<URL>/auth/github.com/callback
      settingsUrl: https://github.com/organizations/openapiary/settings/applications/<id>
    description: ""
    icon: ""
```

Remove one of the ones marked `required` in there and the validation should fail

## Release Notes

```release-note
[installer]: fix the auth provider config
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **mrsimonemms** after the PR has been reviewed.

Associated issue: *#6750*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: installer/pkg/config/v1/config.go
Comment (kylos101) [inline installer/pkg/config/v1/config.go:84]: Cool, looks like this will "dive" into the next level (each member) and trigger its related validator.

File: installer/pkg/config/v1/config.go
Comment (kylos101) [inline installer/pkg/config/v1/config.go:255]: In looking at the config map for core-dev, the `SettingsUrl` does not exist for BitBucket. 

Is it worth using [a custom validator](https://github.com/go-playground/validator/issues/494), to only make this required if the type is GitHub or GitLab?

**Review by kylos101 — COMMENTED** [review]
@MrSimonEmms just a question Re: conditional validation support in Go - thanks for the heads up! Let me know your preference?

File: installer/pkg/config/v1/config.go
Comment (MrSimonEmms) [reply installer/pkg/config/v1/config.go:84]: Correct. We don't need to make `AuthProviders` required (if not set, it'll allow users to use the GUI version of this). The `dive` property [tells it to interrogate the slice](https://github.com/go-playground/validator/blob/master/_examples/simple/main.go#L16)

File: installer/pkg/config/v1/config.go
Comment (MrSimonEmms) [reply installer/pkg/config/v1/config.go]: Good catch. It's not actually a required property according to the [TypeScript interface](https://github.com/gitpod-io/gitpod/blob/b54521de75b143db4a289918b2a9d4a9e96ffca8/components/gitpod-protocol/src/protocol.ts#L1152)

File: installer/pkg/config/v1/config.go
Comment (kylos101) [reply installer/pkg/config/v1/config.go:84]: Perfect, thank you!

File: installer/pkg/config/v1/config.go
Comment (kylos101) [reply installer/pkg/config/v1/config.go]: Gotcha, thank you very much for also sharing the corresponding typescript interface! :100:

**Review by kylos101 — APPROVED** [review]
@MrSimonEmms nice work, amigo! Have a good rest of your evening. :night_with_stars:

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: bd18bf85faaa29856689d38751997e64be0490bc</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6753#pullrequestreview-808882937" title="Approved">kylos101</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **mrsimonemms** after the PR has been reviewed.

Associated issue: *#6750*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6753#issuecomment-971785754" title="Approved">kylos101</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **mrsimonemms** after the PR has been reviewed.

Associated issue: *#6750*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6753#issuecomment-971793760" title="Approved">kylos101</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **mrsimonemms** after the PR has been reviewed.

Associated issue: *#6750*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6753#issuecomment-971793760" title="Approved">kylos101</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/6753#issuecomment-971795203" title="Approved">MrSimonEmms</a>*

Associated issue: *#6750*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)~~ [MrSimonEmms]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
